# HG changeset patch
# User Torsten <ttl@justmail.de>
# Date 1368883023 -7200
#      Sat May 18 15:17:03 2013 +0200
# Node ID 518dbe0ae857a845011cedf1b10af43110d7b405
# Parent  f6dfc77056238cbda3a35fe243ce136f8f761721
fix auto completion for octave files in editor

* file-editor-tab.cc(update_lexer): get keyword list from lexer for auto
  completion

diff --git a/libgui/src/m-editor/file-editor-tab.cc b/libgui/src/m-editor/file-editor-tab.cc
--- a/libgui/src/m-editor/file-editor-tab.cc
+++ b/libgui/src/m-editor/file-editor-tab.cc
@@ -254,18 +254,36 @@ file_editor_tab::update_lexer ()
         }
       else
         {
           // FIXME -- why should the bash lexer be the default?
           lexer = new QsciLexerBash ();
         }
     }
 
+  if (lexer)
+    {
+      QsciAPIs *apis = new QsciAPIs(lexer);
+      if (apis)
+        {
+          QString keyword;
+          QStringList keyword_list;
+          int i;
+          for (i=1; i<=3; i++) // load the first 3 keyword sets
+            {
+              keyword = QString(lexer->keywords (i));           // get list
+              keyword_list = keyword.split (QRegExp ("\\s+"));  // split
+              for (i = 0; i < keyword_list.size (); i++)        // add to API
+                apis->add (keyword_list.at (i));
+            }
+          apis->prepare ();
+        }
+    }
+
   QSettings *settings = resource_manager::get_settings ();
-
   if (settings)
     lexer->readSettings (*settings);
 
   _edit_area->setLexer (lexer);
 
 }
 
 // slot for fetab_set_focus: sets the focus to the current edit area
